nixos/modules/services/misc/redlib.nix

```diff
@@ -41,50 +67,51 @@ in
         description = "Open ports in the firewall for the redlib web interface";
       };
 
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType =
+            with types;
+            attrsOf (
+              nullOr (oneOf [
+                bool
+                int
+                str
+              ])
+            );
+          options = { };
+        };
+        default = { };
+        description = ''
+          See [GitHub](https://github.com/redlib-org/redlib/tree/main?tab=readme-ov-file#configuration) for available settings.
+        '';
+      };
     };
   };
 
   config = mkIf cfg.enable {
+    systemd.packages = [ cfg.package ];
     systemd.services.redlib = {
-        description = "Private front-end for Reddit";
-        wantedBy = [ "multi-user.target" ];
-        after = [ "network.target" ];
-        serviceConfig = {
-          DynamicUser = true;
-          ExecStart = "${lib.getExe cfg.package} ${args}";
-          AmbientCapabilities = lib.mkIf (cfg.port < 1024) [ "CAP_NET_BIND_SERVICE" ];
-          Restart = "on-failure";
-          RestartSec = "2s";
-          # Hardening
-          CapabilityBoundingSet = if (cfg.port < 1024) then [ "CAP_NET_BIND_SERVICE" ] else [ "" ];
-          DeviceAllow = [ "" ];
-          LockPersonality = true;
-          MemoryDenyWriteExecute = true;
-          PrivateDevices = true;
-          # A private user cannot have process capabilities on the host's user
-          # namespace and thus CAP_NET_BIND_SERVICE has no effect.
-          PrivateUsers = (cfg.port >= 1024);
-          ProcSubset = "pid";
-          ProtectClock = true;
-          ProtectControlGroups = true;
-          ProtectHome = true;
-          ProtectHostname = true;
-          ProtectKernelLogs = true;
-          ProtectKernelModules = true;
-          ProtectKernelTunables = true;
-          ProtectProc = "invisible";
-          RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
-          RestrictNamespaces = true;
-          RestrictRealtime = true;
-          RestrictSUIDSGID = true;
-          SystemCallArchitectures = "native";
-          SystemCallFilter = [ "@system-service" "~@privileged" "~@resources" ];
-          UMask = "0077";
-        };
+      wantedBy = [ "default.target" ];
+      environment = mapAttrs (_: v: if isBool v then boolToString' v else toString v) cfg.settings;
+      serviceConfig = {
```

Comment:
I'm confused by the systemd unit change. That doesn't seem like a thing we want to do. Was it broken? Otherwise I would hardly vote for keeping the hardening.

Reply:
The upstream unit contains the hardening, I think using `systemd.packages` would make the code cleaner.